### PR TITLE
APP-6696 pass granular constraints when uploading module from CLI

### PR DESCRIPTION
**Pull request by abe-winter**
## What changed
- add `--tags` field to `module upload` command in CLI
## Why
For precisely targeting module artifacts to platforms (i.e. specific distro or version, not just `linux/amd64`)

### CODE REVIEW

**Review by benjirewis — COMMENTED** [review]
Seems fine, just a general question. Couldn't find the associated scope doc, my apologies.

File: cli/app.go
Comment (abe-winter) [reply cli/app.go]: fixed in c9c4c16ac. new output:

```
   --platform value  platform of the binary you are uploading. Must be one of:
                      any           (most Python modules)
                      any/amd64     (most Docker-based modules)
                      any/arm64
                      linux/any     (Python modules that also require OS support)
                      darwin/any
                      linux/amd64
                      linux/arm64
                      linux/arm32v7
                      linux/arm32v6
                      darwin/amd64  (Intel macs)
                      darwin/arm64  (Apple silicon macs)
   --public-namespace value  the public namespace where the module resides (alternative way of specifying the org id)
   --tags value              Optional extra fields for constraining the platforms to which this binary
                             is deployed. Examples: distro:debian, distro:ubuntu, os_version:22.04,
                             os_codename:jammy. You can provide multiple tags in this field by separating
                             them with a comma. For a machine to use an upload, all tags must be satisified
                             as well as the --platform field.
   --version value  version of the module to upload (semver2.0) ex: "0.1.0"
```

**Review by benjirewis — APPROVED** [review]
LGTM once api changes merged.

File: cli/app.go
Comment (benjirewis) [reply cli/app.go]: Discussed offline with @abe-winter ; thanks for pointers to other PRs + scope.

**Pull request merged**
